### ISSUE TITLE
Change `CaloCalTopoClusters` to `egammaClusters` for PHYSLITE

### ISSUE THREAD

**Issue by ekourlit** [scikit-hep/coffea#975]
**Describe the bug**
There has been some evolution on ATLAS PHYSLITE data format. The branch `CaloCalTopoClusters` has changed to `egammaClusters`. This means the `caloClusters` methods of the [`physlite.py`](https://github.com/CoffeaTeam/coffea/blob/master/src/coffea/nanoevents/methods/physlite.py) need to be updated. The relevant methods can be found here: [1](https://github.com/CoffeaTeam/coffea/blob/master/src/coffea/nanoevents/methods/physlite.py#L250) and [2](https://github.com/CoffeaTeam/coffea/blob/master/src/coffea/nanoevents/methods/physlite.py#L256).

**To Reproduce**
If you try to request the `caloClusters` on an object, e.g.:
```python
events.Electrons.caloClusters
```
it will fail with error:
`FieldNotFoundError: no field 'CaloCalTopoClusters' in record with 1 fields`

**Expected behavior**
I would expect to look for the branch `egammaClusters` instead of `CaloCalTopoClusters`.

**Desktop (please complete the following information):**
 - OS: macOS
 - Version: 14.2.1
 - Coffea version: 2023.12.0

**Additional context**
Tagging @nikoladze , @matthewfeickert

**Comment by nikoladze** [comment]
we could consider making this a bit less hard coded. In principle the naming scheme tells us that everything ending in "Link" or "Links" is going to be an ElementLink. Also we can read the target collection hash to name mapping from the MetaData/EventFormat branch (unfortunately needs custom code to read). On the other hand we don't want to do that for every file we process. So maybe something like
- have a default list of ElementLink fields with their corresponding targets (e.g. based on most recent PHYSLITE version)
- have a method to change/pass this to the schema
- have a method to generate it from reading a single file

Of course this only covers links to a single target (most of the cases except MET, Truth). The others are currently broken and would need a different handling in dask anyway.

**Comment by matthewfeickert** [comment]
> The relevant methods can be found here

(for posterirty and ease of viewing, here are permenant links)

https://github.com/CoffeaTeam/coffea/blob/5bc65a561b299f3a6b7f270e1b0b02deafe495b3/src/coffea/nanoevents/methods/physlite.py#L250-L253

https://github.com/CoffeaTeam/coffea/blob/5bc65a561b299f3a6b7f270e1b0b02deafe495b3/src/coffea/nanoevents/methods/physlite.py#L256-L259

**Comment by lgray** [comment]
Hmm, I need to do a similar bit of automation for the edm4hep schema. People like to get waaay too clever with self-describing data structures and it hurts usability/performance.

**Comment by matthewfeickert** [comment]
I think this is fixed via PR https://github.com/CoffeaTeam/coffea/pull/1156? Though we didn't have a minimal failing example in this issue, so we'll need to get an example to verify this.

**Comment by matthewfeickert** [comment]
Yeah, this Issue can get closed as it was resolved in PR https://github.com/CoffeaTeam/coffea/pull/1156.

Using the test file at `tests/samples/PHYSLITE_example.root`

```python
from pathlib import Path
from coffea.nanoevents import NanoEventsFactory, PHYSLITESchema

file_path = Path().cwd() / "tests" / "samples" / "PHYSLITE_example.root"

events = NanoEventsFactory.from_root(
    {file_path: "CollectionTree"},
    schemaclass=PHYSLITESchema,
).events()

print(f"# events.Electrons.caloClusters={events.Electrons.caloClusters}")
# events.Electrons.caloClusters=dask.awkward<apply-global-index, npartitions=1>
print(f"# events.Electrons.caloClusters.comput() = {events.Electrons.caloClusters.compute()}")
# events.Electrons.caloClusters.comput() = [[[{calE: 1.38e+05, _eventindex: 0, calEta: 1.55, calPhi: ..., ...}], ...], ...]
```

**Comment by lgray** [comment]
Fixed by #1156
